lib/httparty/request/body.rb

```diff
@@ -46,26 +46,16 @@ def generate_multipart
         multipart += "--#{boundary}--\r\n"
       end
 
-      def has_file?(hash)
-        hash.detect do |key, value|
-          if value.respond_to?(:to_hash) || includes_hash?(value)
-            has_file?(value)
-          elsif value.respond_to?(:to_ary)
-            value.any? { |e| file?(e) }
-          else
-            file?(value)
-          end
+      def file?(value)
+        if value.respond_to?(:to_hash)
+          value.to_hash.any? { |_, v| file?(v) }
+        elsif value.respond_to?(:to_ary)
+          value.to_ary.any? { |v| file?(v) }
+        else
+          value.respond_to?(:path) && value.respond_to?(:read)
```

Comment:
Previous use of a method was more readable and understandable , imho.
It showed that if value responds to `path` and `read` it is a file, which is exactly what we are looking for.

Reply:
you mean this method?

```ruby
      def file?(object)
        object.respond_to?(:path) && object.respond_to?(:read)
      end
```

if that's what you meant, i think it makes sense, and also would help for the `generate_multipart` which should only check if the value is a file (not recursive) if i understand it correctly.

I changed it here as well: 5a778ed2eac655ecebc0e2b134e9d53a29026aa9